meteo.go

```diff
@@ -0,0 +1,52 @@
+package meteo
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+const (
+	userAgent = "Meteo/0.1 https://github.com/qba73/meteo"
+)
+
+// Weather represents weather conditions
+// in a geographical region.
+type Weather struct {
+	Summary string
+	Temp    float64
+}
+
+// String implements stringer interface.
+func (w Weather) String() string {
+	return fmt.Sprintf("%s %.1f°C", strings.Title(w.Summary), w.Temp)
+}
+
+// NameResolver interface is used by an Meteo Client
+// to obtain geo coordinates for given place located in
+// a country identified by country id.
+type NameResolver interface {
+	// GetCoordinates takes place and country code
+	// and returns geo information like lat and lng.
+	GetCoordinates(placeName, country string) (Place, error)
+}
+
+// RunCLI is a main function that runs the cli machinery.
+func RunCLI() {
+	uname := os.Getenv("GEO_USERNAME")
+	resolver, err := NewWikipediaClient(uname)
```

Comment:
Something a little unclear here—`NewWikipediaClient` doesn't tell us this is about geolocation resolving. And `NewNorwayClient` doesn't tell us that's about weather.

What about something like:

```suggestion
	c, err := NewYrWeatherClient(
		WithWikipediaGeoResolver(),
	)
```

(and `WithWikipediaGeoResolver` would just look up the username in the environment)